app/Models/User.php

```diff
@@ -73,6 +78,26 @@ class User extends Authenticatable
         'password_expire_at' => 'immutable_datetime',
     ];
 
+    protected static function fieldValidator(string $field): string|array
+    {
+        return match ($field) {
+            'name' => 'required|string',
+            'email' => 'nullable|string|email',
+            'has_email_verified' => 'required_with:email|bool',
+            'password' => [
+                'bail',
+                'nullable',
+                'string',
+                'different:current',
+                Password::min(8)->letters()->mixedCase()->numbers()->uncompromised(),
+            ],
+            'password_expire_at' => 'required_with:password|nullable|date',
```

Comment:
w sumie to nie zadziała tak jakbym chciał

chodzi o to, żebym normalnie mógł zapisać datę lub null, null czyli hasło nie wygasa nigdy

nullable|required nie pozwala wpisać null :(

tak, wiem, tak napisałem w zadaniu :D

powinno być present_with, ale akurat laravel nie ma takiej walidacji lol, ma tylko present albo missing_with

Reply:
dodałem jakiegoś custom rula, niby działa, ale nie wiem czy w poprawnej formie komunikat błędu walidacji zwróci